ibis/expr/analysis.py

```diff
@@ -1039,75 +1037,69 @@ def is_scalar_reduction(expr):
     return isinstance(expr, ir.Scalar) and is_reduction(expr)
 
 
-class _AnyToExistsTransform:
-    __slots__ = (
-        "expr",
-        "parent_table",
-        "query_roots",
-        "foreign_table",
-        "predicates",
-    )
+_ANY_OP_MAPPING = {
+    ops.Any: ops.UnresolvedExistsSubquery,
+    ops.NotAny: ops.UnresolvedNotExistsSubquery,
+}
 
-    def __init__(self, expr: ir.Expr, parent_table: ir.Table) -> None:
-        self.expr = expr
-        self.parent_table = parent_table
-        self.query_roots: AbstractSet[ops.TableNode] = frozenset(
-            self.parent_table.op().root_tables()
+
+def _make_any(
+    expr,
+    any_op_class: type[ops.Any] | type[ops.NotAny],
+):
+    transform = _AnyToExistsTransform(expr)
+    tables, predicates = transform.get_result()
+
+    ntables = len(tables)
+    if (ntables := len(tables)) == 1:
+        op = any_op_class(expr)
+    else:
+        assert ntables > 0, "no tables found"
+        op = _ANY_OP_MAPPING[any_op_class](
+            tables=tables,
+            predicates=predicates,
         )
-        self.foreign_table: ir.Table | None = None
-        self.predicates: MutableSequence[ir.BooleanValue] = []
+    return op.to_expr()
 
-    def get_result(self) -> ir.BooleanColumn:
-        self._visit(self.expr)
 
-        op = self.expr.op()
+class _AnyToExistsTransform:
```

Comment:
How about removing this class entirely?

https://github.com/ibis-project/ibis/commit/38738d488cc0c66b4a4444a42ac567f33e7658ef

Reply:
Rolled this in. Amazing. 🙇🏻